File: Leetcode/C++/Array/disap.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& arr) {
        unordered_set<int>s;
        vector<int>res;
        int n=arr.size();
        vector<int>ans;
        for(int i=0;i<arr.size();i++)
        {
            if(s.find(arr[i])==s.end())
            {
                s.insert(arr[i]);
                ans.push_back(arr[i]);
            } 
            
               
        }
        int size=ans.size();
        int need=n-size;
        for(int i=1;i<=size+need;i++)
        {
            if(s.find(i)==s.end())
            {
                res.push_back(i);
            }
        }
        return res;
    }
};

File: Leetcode/Java/Array/disap.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
       Set<Integer> seen = new HashSet<>();

       for(int num: nums){
        seen.add(num);
       }
Arrays.sort(nums);
List<Integer> ans = new ArrayList<>();

       for(int i=0;i <nums.length; i++){
            if((i+1)!=nums[i] && !seen.contains(i+1)){
                ans.add(i+1);
            }
       }

       return ans;
    }
}
